geode-core/src/main/java/org/apache/geode/internal/util/DriverJarUtil.java

```diff
@@ -0,0 +1,158 @@
+      throws ClassNotFoundException, IllegalAccessException,
+      InstantiationException, SQLException {
+    URLClassLoader urlClassLoader = createUrlClassLoader();
+    Driver driver = getDriverClassByName(driverClassName, urlClassLoader);
+    Driver d = new DriverWrapper(driver);
+    registerDriverWithDriverManager(d);
+  }
+
+
+  public String getJdbcDriverName(String driverJarName) throws IOException {
+    return getJdbcDriverName(createDeployedJar(driverJarName));
+  }
+
+  public String getJdbcDriverName(DeployedJar jar) throws IOException {
+    File jarFile = jar.getFile();
+    FileInputStream fileInputStream = createFileInputStream(jarFile.getAbsolutePath());
+    BufferedInputStream bufferedInputStream = createBufferedInputStream(fileInputStream);
+    ZipInputStream zipInputStream = createZipInputStream(bufferedInputStream);
+    ZipEntry zipEntry;
+    while ((zipEntry = zipInputStream.getNextEntry()) != null) {
+      // JDBC 4.0 Drivers must include the file META-INF/services/java.sql.Driver. This file
+      // contains the name of the JDBC drivers implementation of java.sql.Driver
+      // See https://docs.oracle.com/javase/8/docs/api/java/sql/DriverManager.html
+      if (!zipEntry.getName().equals("META-INF/services/java.sql.Driver")) {
+        continue;
+      }
+      int size = (int) zipEntry.getSize();
+      if (size == -1) {
+        throw new IOException("Invalid zip entry found for META-INF/services/java.sql.Driver " +
+            "within jar. Ensure that the jar containing the driver has been deployed and that " +
+            "the driver is at least JDBC 4.0");
+      }
+      byte[] bytes = new byte[size];
+      int offset = 0;
+      int chunk;
+      while ((size - offset) > 0) {
+        chunk = zipInputStream.read(bytes, offset, size - offset);
+        if (chunk == -1) {
+          break;
+        }
+        offset += chunk;
+      }
+      return new String(bytes);
+    }
+    throw new IOException("Could not find JDBC Driver class name in jar file '"
+        + jar.getJarName() + "'");
+  }
+
+  // The methods below are included to facilitate testing and to make the helper methods in this
+  // class cleaner
+  FileInputStream createFileInputStream(String jarFilePath) throws FileNotFoundException {
+    return new FileInputStream(jarFilePath);
+  }
+
+  BufferedInputStream createBufferedInputStream(FileInputStream fileInputStream) {
+    return new BufferedInputStream(fileInputStream);
+  }
+
+  ZipInputStream createZipInputStream(BufferedInputStream bufferedInputStream) {
+    return new ZipInputStream(bufferedInputStream);
+  }
+
+  DeployedJar createDeployedJar(String driverJarName) throws IOException {
+    return ClassPathLoader.getLatest().getJarDeployer()
+        .findLatestValidDeployedJarFromDisk(driverJarName);
+  }
+
+  URLClassLoader createUrlClassLoader() {
+    return new URLClassLoader(ClassPathLoader.getLatest().getJarDeployer().getDeployedJarURLs());
```

Comment:
Please don't rely on having access to the actual jar. The `ClassPathLoader` is probably going to be changing in how it currently works such that if you do the following:
```
deploy --jar=mysql-driver.jar
deploy --jar=my-awesome-app.jar
```
The above is not guaranteed to return `mysql-driver.jar`

You should really only be dealing with the class name of your driver.

Reply:
We've modified this to move away from the URL class loader and just invoke Class.forName on the class name instead, so that should solve the problem of using the list of all deployed Jars here.